src/app/3d/qgs3dmapcanvaswidget.h

```diff
@@ -140,20 +141,21 @@ class APP_EXPORT Qgs3DMapCanvasWidget : public QWidget
     QTimer *mLabelNavSpeedHideTimeout = nullptr;
     Qgs3DMapToolIdentify *mMapToolIdentify = nullptr;
     Qgs3DMapToolMeasureLine *mMapToolMeasureLine = nullptr;
-    Qgs3DMapToolPointCloudChangeAttribute *mMapToolPointCloudChangeAttribute = nullptr;
+    std::unique_ptr<Qgs3DMapToolPointCloudChangeAttribute> mMapToolChangeAttribute;
```

Comment:
Qgs3DMapTool is QObject-based, and unique_ptr is not the best fit unfortunately... because QObject-based objects can be deleted by their parents. There is `QObjectUniquePtr` defined in qgis_core library that should work as unique_ptr, and be aware of QObject as well